modules/articles/server/controllers/articles.server.controller.js

```diff
@@ -30,7 +30,14 @@ exports.create = function (req, res) {
  * Show the current article
  */
 exports.read = function (req, res) {
-  res.json(req.article);
+  // convert mongoose document to JSON
+  var article = req.article ? req.article.toJSON() : {};
+
+  // Add a custom field to the Article, for determining if the current User is the "owner".
+  // NOTE: This field is NOT persisted to the database, since it doesn't exist in the Article model.
+  article.isCurrentUserOwner = req.user && article.user && article.user._id.toString() === req.user._id.toString() ? true : false;
```

Comment:
@vaucouleur I think the way it's written out is more clear. Especially since we need to verify we have a `req.user` object as well. But I agree it's not the most elegant way to do it.